solutions/swb-reference/src/services/keyPairService.test.ts

```diff
@@ -0,0 +1,174 @@
+  SendPublicKeyRequest,
+  DatabaseError,
+  NonUniqueKeyError,
+  NoKeyExistsError
+} from '@aws/swb-app';
+import { DynamoDBService, JSONValue } from '@aws/workbench-core-base';
+import PaginatedJsonResponse from '@aws/workbench-core-base/lib/interfaces/paginatedJsonResponse';
+import KeyPairService from './keyPairService';
+
+describe('KeyPairService', () => {
+  const region = 'us-east-1';
+  const dynamoDBService = {} as DynamoDBService;
+  const keyPairService: KeyPairService = new KeyPairService(dynamoDBService);
+
+  beforeAll(() => {
+    process.env.AWS_REGION = region;
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('getKeyPair', () => {
+    let getKeyPairRequest: GetKeyPairRequest;
+
+    beforeEach(() => {
+      getKeyPairRequest = { projectId: '' };
+    });
+
+    test('should throw not implemented error', async () => {
+      await expect(() => keyPairService.getKeyPair(getKeyPairRequest)).rejects.toThrow(
+        new Error('Method not implemented.')
+      );
+    });
+  });
+
+  describe('listKeyPairs', () => {
+    let listKeyPairsRequest: ListKeyPairsRequest;
+
+    beforeEach(() => {
+      listKeyPairsRequest = { userId: '' };
+    });
+
+    test('should throw not implemented error', async () => {
+      await expect(() => keyPairService.listKeyPairs(listKeyPairsRequest)).rejects.toThrow(
+        new Error('Method not implemented.')
+      );
+    });
+  });
+
+  describe('deleteKeyPair', () => {
+    let deleteKeyPairRequest: DeleteKeyPairRequest;
+
+    beforeEach(() => {
+      deleteKeyPairRequest = { projectId: 'proj-123', userId: 'user-123' };
+    });
+
+    describe('when no key exists', () => {
+      beforeEach(() => {
+        const queryResponse = { data: [] };
+        dynamoDBService.getPaginatedItems = jest.fn((): Promise<PaginatedJsonResponse> => {
+          return Promise.resolve(queryResponse);
+        });
+      });
+
+      test('it throws NoKeyExistsError', async () => {
+        // OPERATE n CHECK
+        await expect(() => keyPairService.deleteKeyPair(deleteKeyPairRequest)).rejects.toThrow(
+          NoKeyExistsError
+        );
+      });
+    });
+
+    describe('when no unique key exists', () => {
+      beforeEach(() => {
+        const queryResponse = {
+          data: [
+            {
+              pk: 'SSH#ssh-123',
+              sk: 'USER#user-123',
+              dependency: 'proj-123'
+            },
+            { pk: 'SSH#ssh-456', sk: 'USER#user-123', dependency: 'proj-123' }
+          ]
+        };
+        dynamoDBService.getPaginatedItems = jest.fn((): Promise<PaginatedJsonResponse> => {
+          return Promise.resolve(queryResponse);
+        });
+      });
+
+      test('it throws NonUniqueKeyQuery', async () => {
+        // OPERATE n CHECK
+        await expect(() => keyPairService.deleteKeyPair(deleteKeyPairRequest)).rejects.toThrow(
+          NonUniqueKeyError
+        );
+      });
+    });
+
+    describe('when key exists', () => {
```

Comment:
might want to call out that we're only expecting a single key here.

Reply:
Good call! I am all for increased clarity!